### PR TITLE
feat: api references

### PR DESCRIPTION
**Pull request by suhailkakar**
## Description

Added the new API references page with custom parsers and component to render APIs into the docs.

### CODE REVIEW

File: generate-api-reference.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+const fs = require('fs');
+const yaml = require('js-yaml');
+
+const API_SCHEMA_URL =
+  'https://raw.githubusercontent.com/suhailkakar/livepeer-studio/master/packages/api/src/schema/api-schema.yaml';
```
Comment (adamsoffer) [inline generate-api-reference.js:5]: Will we need to update this with the production repo?

File: pages/reference/api/_meta.en-US.json [hunk 2]
```diff
@@ -0,0 +1,166 @@
+{
+  "create-a-stream": {
```
Comment (adamsoffer) [inline pages/reference/api/_meta.en-US.json:2]: Will we have to maintain this list? It won't get autogenerated based on the schema?

**Review by adamsoffer — CHANGES_REQUESTED** [review]

**Review by adamsoffer — APPROVED** [review]

File: pages/reference/api/_meta.en-US.json (hunk 2 above)
Comment (suhailkakar) [reply pages/reference/api/_meta.en-US.json:2]: Nope, this list is automatically generated here: https://github.com/livepeer/docs/blob/c43e4854280d69c2c0c822c5bd065616fd9e1461/generate-api-reference.js#LL45C1-L56C2

File: generate-api-reference.js (hunk 1 above)
Comment (suhailkakar) [reply generate-api-reference.js:5]: Yes, I am waiting for this pr to get merged: https://github.com/livepeer/studio/pull/1754

File: pages/reference/api/_meta.en-US.json (hunk 2 above)
Comment (suhailkakar) [reply pages/reference/api/_meta.en-US.json:2]: Yes, this list is automatically generated here: https://github.com/livepeer/docs/blob/c43e4854280d69c2c0c822c5bd065616fd9e1461/generate-api-reference.js#LL45C1-L56C2

**Pull request merged**